src/Uno.UI.Composition/Composition/ColorBrushTransitionState.skia.cs

```diff
@@ -0,0 +1,41 @@
+﻿#nullable enable
+
+using Windows.UI;
+
+namespace Microsoft.UI.Composition;
+
+internal sealed class ColorBrushTransitionState
+{
+	internal ColorBrushTransitionState(BorderVisual visual, Color fromColor, Color toColor, long startTimestamp, long endTimestamp)
+	{
+		Visual = visual;
+		FromColor = fromColor;
+		ToColor = toColor;
+		StartTimestamp = startTimestamp;
+		EndTimestamp = endTimestamp;
+	}
+
+	internal BorderVisual Visual { get; }
+	internal Color FromColor { get; set; }
+	internal Color ToColor { get; set; }
+	internal long StartTimestamp { get; set; }
+	internal long EndTimestamp { get; set; }
+
```

Comment:
I wonder if these should not have setters?

Reply:
The setters are used in case the background is updated while a transition is in progress

Reply:
I strongly disagree with this. I would much rather keep ColorBrushTransitionState immutable and create a new instance when the background is updated instead of rewriting the properties.